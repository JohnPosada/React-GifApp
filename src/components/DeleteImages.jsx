
import { AiFillDelete } from "react-icons/ai";

export const DeleteImages = ( { onDelete }) => {

  return (
    <AiFillDelete className="delete" size= '35px' onClick={onDelete}/>)
}
