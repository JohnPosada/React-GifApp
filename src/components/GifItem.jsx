
import PropTypes from "prop-types"

export const GifItem = ({title, image}) => {


  return (
    <div className="card">
        <img src={image} alt={title}></img>
        <p>{title}</p>
    </div>
  )

  
}


GifItem.propTypes = {
  title: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired
}

