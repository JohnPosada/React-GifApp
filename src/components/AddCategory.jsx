import { useState } from "react"
import { DeleteImages } from "./DeleteImages";

export const AddCategory = ( { onNewCategory, onDelete, categories }) => {

  const [inputValue, setInputValue] = useState('');

  const onInputChange= (event) => {
    setInputValue(event.target.value);
  }

  const onSubmit= (event) => {
    event.preventDefault();
    if (inputValue.trim().length <=1) return ;
    //setCategories(categories => [...categories, inputValue]);
    onNewCategory(inputValue.trim())
    setInputValue('');
  }

  return (
    <form className="display: flex" onSubmit={ (event) => onSubmit(event) }>

      <input 
          type="text"
          placeholder='buscar gifs'
          value={inputValue}
          onChange={ (event) => onInputChange(event) }
      />
      {
        categories.length!=0 && (<DeleteImages onDelete= {onDelete} />)
      }
    </form>
  )
}
