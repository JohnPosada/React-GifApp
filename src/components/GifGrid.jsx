
import {GifItem} from './GifItem.jsx'
import { useFetchGifs } from '../hooks/useFetchGifs.js'


export const GifGrid = ({category}) => {

  const {images, isloading} = useFetchGifs(category);

  return (<>
    <h2> {category} </h2>

    {
      isloading && (<p> Cargando...</p>)
    }

    <div className='card-grid'>
      {
        images.map((img) => {
          return <GifItem 
            key={img.id}
            {...img}

            />
      })}
    </div>
  </>   
  )
}
