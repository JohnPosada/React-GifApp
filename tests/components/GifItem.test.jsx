import { render, screen } from "@testing-library/react";
import {GifItem} from "../../src/components/GifItem";


describe('Pruebas a GifItem', () => { 
    
    const title= 'anime scenery GIF';
    const url= 'anime.jpg';


    test('Debe hacer match con el snapshot', () => { 

        const { container } = render( <GifItem title={ title } image={ url } /> );
        expect( container ).toMatchSnapshot();

     })

     test('Debe de mostrar la imagen con el scr y el Alt especificado', () => { 
        render( <GifItem title={ title } image={ url } /> );
        const { src, alt } = screen.getByRole('img');
        expect(src).toBe(`http://localhost/${url}`)
        expect(alt).toBe(title)

      })
 })