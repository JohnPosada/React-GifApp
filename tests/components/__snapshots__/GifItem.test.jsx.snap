// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba a GifItem Debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="anime scenery GIF"
      src="anime.jpg"
    />
    <p>
      anime scenery GIF
    </p>
  </div>
</div>
`;
